app/org/sagebionetworks/bridge/dynamodb/DynamoNotificationTopicDao.java

```diff
@@ -0,0 +1,146 @@
+import com.amazonaws.services.sns.model.CreateTopicResult;
+import com.amazonaws.services.sns.model.DeleteTopicRequest;
+
+@Component
+public class DynamoNotificationTopicDao implements NotificationTopicDao {
+    private static Logger LOG = LoggerFactory.getLogger(DynamoNotificationTopicDao.class);
+    
+    private DynamoDBMapper mapper;
+    
+    private AmazonSNSClient snsClient;
+    
+    private BridgeConfig config;
+    
+    @Resource(name = "notificationTopicMapper")
+    final void setNotificationTopicMapper(DynamoDBMapper mapper) {
+        this.mapper = mapper;
+    }
+    
+    @Resource(name = "snsClient")
+    final void setSnsClient(AmazonSNSClient snsClient) {
+        this.snsClient = snsClient;
+    }
+
+    @Resource(name = "bridgeConfig")
+    public void setBridgeConfig(BridgeConfig bridgeConfig) {
+        this.config = bridgeConfig;
+    }
+    
+    @Override
+    public List<NotificationTopic> listTopics(StudyIdentifier studyId) {
+        checkNotNull(studyId);
+        
+        DynamoNotificationTopic hashKey = new DynamoNotificationTopic();
+        hashKey.setStudyId(studyId.getIdentifier());
+        DynamoDBQueryExpression<DynamoNotificationTopic> query = new DynamoDBQueryExpression<DynamoNotificationTopic>()
+                .withConsistentRead(false).withHashKeyValues(hashKey);
+
+        QueryResultPage<DynamoNotificationTopic> resultPage = mapper.queryPage(DynamoNotificationTopic.class, query);
+        return resultPage.getResults().stream().map(obj -> (NotificationTopic) obj).collect(toImmutableList());
+    }
+
+    @Override
+    public NotificationTopic getTopic(StudyIdentifier studyId, String guid) {
+        checkNotNull(studyId);
+        checkNotNull(guid);
+        
+        DynamoNotificationTopic hashKey = new DynamoNotificationTopic();
+        hashKey.setStudyId(studyId.getIdentifier());
+        hashKey.setGuid(guid);
+
+        DynamoNotificationTopic topic = mapper.load(hashKey);
+        if (topic == null) {
+            throw new EntityNotFoundException(NotificationTopic.class);
+        }
+        return topic;
+    }
+
+    @Override
+    public NotificationTopic createTopic(NotificationTopic topic) {
+        checkNotNull(topic);
+        
+        // Create SNS topic first. If SNS fails, an exception is thrown. If DDB call fails, the SNS topic is orphaned
+        // but that will not break the data integrity of Bridge data.
+        
+        topic.setGuid(BridgeUtils.generateGuid());
+        
+        String snsTopicName = createSnsTopicName(topic);
+        CreateTopicRequest request = new CreateTopicRequest().withName(snsTopicName);
+        CreateTopicResult result = snsClient.createTopic(request);
+        topic.setTopicARN(result.getTopicArn());
+        
+        mapper.save(topic);
+        return topic;
+    }
+
+    @Override
+    public NotificationTopic updateTopic(StudyIdentifier studyId, NotificationTopic topic) {
```

Comment:
That might make sense. The studyId should ultimately come from the user's session so it can't be set to another study, either by carrying it over, or ensuring it is set in the service.